### PR TITLE
Upgrade the rust grpc-client to the ontologies breaking changes - EI-615

### PR DESCRIPTION
**Pull request by mihai-dinculescu**

### CODE REVIEW

File: src/properties.rs [hunk 1]
```diff
@@ -0,0 +1,170 @@
+use thiserror::Error;
+
+use crate::{
+    client::iotics::api::{LangLiteral, Literal, StringLiteral},
+    common::{Property, Uri, Value},
+};
+
+pub mod common_keys {
```
Comment (mihai-dinculescu) [inline src/properties.rs:8]: @Elleppi these constants and the helper functions being here means that we can take them out from the connector engine.

File: src/twin/crud.rs [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+    let mut client = create_twin_api_client(auth_builder.clone()).await?;
+
+    update_twin_with_client(auth_builder, &mut client, did, properties).await
+}
+
+pub async fn update_twin_with_client(
+    auth_builder: Arc<impl IntoAuthBuilder>,
+    client: &mut TwinApiClient<Channel>,
+    did: &str,
+    properties: PropertyUpdate,
```
Comment (mihai-dinculescu) [inline src/twin/crud.rs:133]: @Elleppi this now takes a `PropertyUpdate` object so that it's more generic and flexible.

File: src/twin/crud.rs [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+use crate::auth_builder::IntoAuthBuilder;
+use crate::common::{
+    FeedId, FeedValue, FeedValues, GeoLocation, Headers, Property, PropertyUpdate, TwinId,
+    Visibility,
+};
+use crate::helpers::generate_client_app_id;
+
+use super::{create_feed_api_client, create_twin_api_client, FeedApiClient, TwinApiClient};
+
+pub async fn create_update_twin(
```
Comment (Elleppi) [inline src/twin/crud.rs:30]: Do we actually use these functions or we always use the `*_with_client()` only ?

**Review by Elleppi — COMMENTED** [review]
Overall it looks good. Left some questions

File: src/client/iotics.api.rs [hunk 4]
```diff
@@ -328,11 +314,10 @@ pub mod create_feed_request {
     #[derive(Clone, PartialEq, ::prost::Message)]
     pub struct Payload {
         /// ID of the feed to create
+        ///
+        /// StoreLast indicates if the last received value should be stored of not
```
[the remark this answers is not in the document]
Comment (mihai-dinculescu) [reply src/client/iotics.api.rs:318]: This is code auto-generated from the proto files. @StephanieBracaloni FYI

File: src/client/iotics.api.rs [hunk 5]
```diff
@@ -1150,12 +1090,9 @@ pub mod create_twin_response {
     /// Payload identifies the twin which was created.
     #[derive(Clone, PartialEq, ::prost::Message)]
     pub struct Payload {
-        /// created twin
+        /// Unique ID of the twin to delete
```
[the remark this answers is not in the document]
Comment (mihai-dinculescu) [reply src/client/iotics.api.rs:1093]: This is code auto-generated from the proto files. @StephanieBracaloni FYI

File: src/client/iotics.api.rs [hunk 6]
```diff
@@ -1377,9 +1270,9 @@ pub mod update_twin_response {
     /// UpdateTwinResponse payload.
     #[derive(Clone, PartialEq, ::prost::Message)]
     pub struct Payload {
-        /// Updated Twin
+        /// Unique ID of the twin to delete
```
[the remark this answers is not in the document]
Comment (mihai-dinculescu) [reply src/client/iotics.api.rs:1273]: This is code auto-generated from the proto files. @StephanieBracaloni FYI

File: src/twin/crud.rs (hunk 3 above)
Comment (mihai-dinculescu) [reply src/twin/crud.rs:30]: It really depends on the intended usage.

If called a few times, it's more comfortable to use the version that creates the client internally.
If called many times, it's a lot more performant to use the `*_with_client()` version.

File: src/twin/upsert.rs [hunk 7]
```diff
@@ -1,23 +1,45 @@
...
 
-pub use crate::client::iotics::api::UpsertFeedWithMeta;
-pub use crate::client::iotics::api::UpsertTwinResponse;
+use super::{create_twin_api_client, UpsertFeedWithMeta, UpsertTwinResponse};
 
 #[allow(clippy::too_many_arguments)]
 pub async fn upsert_twin(
+    auth_builder: Arc<impl IntoAuthBuilder>,
+    did: &str,
+    properties: Vec<Property>,
```
[the remark this answers is not in the document]
Comment (mihai-dinculescu) [reply src/twin/upsert.rs:19]: `upsert` always wipes all existing properties, so none of the functionality provided by `PropertyUpdate` would actually work.

**Review by Elleppi — APPROVED** [review]

**Pull request merged**